ffmpeg.py

```diff
@@ -0,0 +1,81 @@
+import shlex
+import subprocess
+import json
+
+from const import qp_str, pts_time_str, y_str, streams_str, x_str, side_data_str, video_stream_str, audio_stream_str, \
+    codec_type_str
+
+
+# query data
+def get_packets_info(file_path):
+    """
+    Get packet data using ffprobe to calculate the bitrate
+    :param file_path input video file path
+    :return information of packet
+    """
+    cmd = 'ffprobe -show_packets -of json'
+    args = shlex.split(cmd)
+    args.append(file_path)
+    output = subprocess.check_output(args, stderr=subprocess.DEVNULL)
+    output = json.loads(output)
+    return output
+
+
+def get_stream_info(file_path):
+    """
+    Get video stream information using ffprobe
+    :param file_path input video file path
+    :return information of stream
+    """
+    cmd = 'ffprobe -show_streams -of json'
+    args = shlex.split(cmd)
+    args.append(file_path)
+    output = subprocess.check_output(args, stderr=subprocess.DEVNULL)
+    output = json.loads(output)
+    streams = output[streams_str]
+    data = {video_stream_str: [], audio_stream_str: []}
+    for stream in streams:
+        data[stream[codec_type_str] + '_stream'].append(stream)
+    return data
+
+
+def generate_vis_video(file_path, folder_path, op):
+    """
+    Generate visualization video from ffmpeg codecview filter using subprocess
+    :param file_path input video file path
+    :param folder_path output folder path
+    :param op option of codecview filter (qp, bs, b_type)
+    """
+    # TODO: change the path to global ffmpeg url
+    cmd = '../FFmpeg/ffmpeg -export_side_data +venc_params -i ' + file_path + ' -vf codecview=' + op + '=true -y'
```

Comment:
You should just use 'ffmpeg'
And check if its in the path at startup and error if not.

Reply:
done